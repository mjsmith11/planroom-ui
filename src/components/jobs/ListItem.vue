<template>
    <router-link :to="link" class="list-group-item-action list-group-item" :class="{'list-group-item-dark': inPast, 'list-group-item-light': !inPast}">
        <div class="row">
            <div class="col-lg-9 col-xl-10">{{job.name}}</div>
            <!--Display div only on large a bigger viewport-->
            <div class="col-lg-3 col-xl-2 d-none d-lg-block"><strong>{{ job.bidDate | date }}</strong></div>
        </div>
    </router-link>
</template>
<script>
import { datetimeFilters } from '../../mixins/datetimeFilters'
export default {
  mixins: [datetimeFilters],
  props: {
    job: Object
  },
  computed: {
    inPast () {
      const today = new Date()
      today.setHours(0, 0, 0, 0)
      const todayUTC = Date.UTC(today.getUTCFullYear(), today.getUTCMonth(), today.getUTCDate(), 0, 0, 0, 0)
      const pieces = this.job.bidDate.split('-')
      const utc = Date.UTC(pieces[0], pieces[1] - 1, pieces[2], 0, 0, 0, 0)
      return utc < todayUTC
    },
    link () {
      return 'jobs/' + this.job.id
    }
  }
}
</script>
<style scoped>
  .list-group-item-light {
    color: #606060;
  }
</style>
