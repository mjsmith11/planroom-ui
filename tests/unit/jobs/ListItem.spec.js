import { mount, RouterLinkStub } from '@vue/test-utils'
import ListItem from '@/components/jobs/ListItem'
import moment from 'moment-timezone'

describe('Job List Item', () => {
  const yesterday = {
    id: 1,
    name: "Job Bidding Yesterday",
    bidDate: moment(new Date()).add(-1, 'days').format('YYYY-MM-DD')
  }

  const today = {
    id: 2,
    name: "Job Bidding Today",
    bidDate: moment(new Date()).format('YYYY-MM-DD')
  } 

  const tomorrow = {
    id: 3,
    name: "Job Bidding Tomorrow",
    bidDate: moment(new Date()).add(1, 'days').format('YYYY-MM-DD')
  }

  it('creates link address', () => {
    let cmp = mount(ListItem, {
      propsData: {
        job: yesterday
      },
      stubs: {
        RouterLink: RouterLinkStub
      }
    })
    expect(cmp.vm.link).toBe('jobs/1')
  })

  it('displays yesterday dark', () => {
    let cmp = mount(ListItem, {
      propsData: {
        job: yesterday
      },
      stubs: {
        RouterLink: RouterLinkStub
      }
    })
    expect(cmp.find('a').classes()).toContain('list-group-item-dark')
    expect(cmp.find('a').classes()).not.toContain('list-group-item-light')
  })

  it('displays today light', () => {
    let cmp = mount(ListItem, {
      propsData: {
        job: today
      },
      stubs: {
        RouterLink: RouterLinkStub
      }
    })
    expect(cmp.find('a').classes()).toContain('list-group-item-light')
    expect(cmp.find('a').classes()).not.toContain('list-group-item-dark')
  })

  it('displays tommorow light', () => {
    let cmp = mount(ListItem, {
      propsData: {
        job: tomorrow
      },
      stubs: {
        RouterLink: RouterLinkStub
      }
    })
    expect(cmp.find('a').classes()).toContain('list-group-item-light')
    expect(cmp.find('a').classes()).not.toContain('list-group-item-dark')
  })

  it('has expected html structure', () => {
    let cmp = mount(ListItem, {
      propsData: {
        job: today
      },
      stubs: {
        RouterLink: RouterLinkStub
      }
    })
    expect(cmp.html()).toMatchSnapshot()
  })
})
