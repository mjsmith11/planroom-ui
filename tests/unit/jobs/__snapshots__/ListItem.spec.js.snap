// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job List Item has expected html structure 1`] = `
<a class="list-group-item-action list-group-item list-group-item-light">
  <div class="row">
    <div class="col-lg-9 col-xl-10">Job Bidding Today</div>
    <div class="col-lg-3 col-xl-2 d-none d-lg-block"><strong>09/25/2018</strong></div>
  </div>
</a>
`;
